src/EmailAddressValidator.php

```diff
@@ -0,0 +1,295 @@
+<?php
+
+namespace ipl\Validator;
+
+use ipl\I18n\Translation;
+
+/**
+ * Validates Email address
+ */
+class EmailAddressValidator extends BaseValidator
+{
+    use Translation;
+
+    /**
+     * @var bool If MX check should be enabled
+     */
+    protected $mx = false;
+
+    /**
+     * @var bool If a deep MX check should be done
+     */
+    protected $deep = false;
+
+    /**
+     * Create a new E-mail address validator with optional options
+     *
+     * @param array $options
+     *
+     * available options
+     *
+     * 'mx'   => If MX check should be enabled, boolean
+     * 'deep' => If a deep MX check should be done, boolean
+     */
+    public function __construct(array $options = [])
+    {
+        if (array_key_exists('mx', $options)) {
+            $this->setMxCheck($options['mx']);
+        }
+
+        if (array_key_exists('deep', $options)) {
+            $this->setDeepMxCheck($options['deep']);
+        }
+    }
+
+    /**
+     * Set MX check
+     *
+     * @param bool $mx
+     * @return $this
+     */
+    public function setMxCheck($mx)
+    {
+        $this->mx = $mx;
+
+        return $this;
+    }
+
+    /**
+     * Set Deep MX check
+     *
+     * @param bool $deep
+     * @return $this
+     */
+    public function setDeepMxCheck($deep)
+    {
+        $this->deep = $deep;
+
+        return $this;
+    }
+
+    /**
+     * Internal method to validate the local part of the email address
+     *
+     * @return boolean
+     */
+    private function validateLocalPart($localPart, $email)
```

Comment:
Please use parameter type and return type declarations. Document the parameters!